target/linux/generic/backport-5.10/811-v6.1-0001-nvmem-core-Fix-memleak-in-nvmem_register.patch

```diff
@@ -1,52 +0,0 @@
-From bd1244561fa2a4531ded40dbf09c9599084f8b29 Mon Sep 17 00:00:00 2001
-From: Gaosheng Cui <cuigaosheng1@huawei.com>
-Date: Fri, 16 Sep 2022 13:04:02 +0100
-Subject: [PATCH] nvmem: core: Fix memleak in nvmem_register()
-
-dev_set_name will alloc memory for nvmem->dev.kobj.name in
-nvmem_register, when nvmem_validate_keepouts failed, nvmem's
-memory will be freed and return, but nobody will free memory
-for nvmem->dev.kobj.name, there will be memleak, so moving
-nvmem_validate_keepouts() after device_register() and let
-the device core deal with cleaning name in error cases.
-
-Fixes: de0534df9347 ("nvmem: core: fix error handling while validating keepout regions")
```

Comment:
I think this patch is still needed, it references this patch: target/linux/generic/backport-5.10/805-v5.15-0003-nvmem-core-fix-error-handling-while-validating-keepo.patch

Reply:
Thanks, @hauke.  I took a stab at modifying it.  Please see the latest push.